packages/dx-react-grid/src/plugins/table-inline-cell-editing.tsx

```diff
@@ -0,0 +1,150 @@
+                  ...row, ...getRowChange(rowChanges, rowId!),
+                };
+
+                const value = getCellValue(changedRow, columnName);
+                const onValueChange = (newValue: any) => {
+                  const changeArgs = {
+                    rowId,
+                    change: createRowChange(changedRow, newValue, columnName),
+                  };
+                  changeRow(changeArgs);
+                };
+                const onKeyDown = (key) => {
+                  if (key === 'Enter') {
+                    commitChangedRows({ rowIds: [rowId] });
+                    stopEditCells({ editingCells: [{ rowId, columnName }] });
+                  } else if (key === 'Escape') {
+                    cancelChangedRows({ rowIds: [rowId] });
+                    stopEditCells({ editingCells: [{ rowId, columnName }] });
+                  }
+                };
+                const onBlur = () => {
+                  commitChangedRows({ rowIds: [rowId] });
+                  stopEditCells({ editingCells: [{ rowId, columnName }] });
+                };
+                const onFocus = selectTextOnEditStart ? e => e.target.select() : () => {};
+                const editingEnabled = isColumnEditingEnabled(columnName);
+
+                return(
+                  <TemplatePlaceholder
+                    name="valueEditor"
+                    params={{
+                      column,
+                      row,
+                      value,
+                      onValueChange,
+                      disabled: !editingEnabled,
+                    }}
+                  >
+                    {content => (
+                      <EditCell
+                        {...params}
+                        row={row}
+                        column={column!}
+                        value={value}
+                        editingEnabled={editingEnabled}
+                        onValueChange={onValueChange}
+                        autoFocus
+                        onKeyDown={({ key }) => onKeyDown(key)}
+                        onBlur={onBlur}
+                        onFocus={onFocus}
+                      >
+                        {content}
+                      </EditCell>
+                    )}
+                  </TemplatePlaceholder>
+                );
+              }
+
+              if (startEditAction !== 'click' && startEditAction !== 'doubleClick') {
+                throw new Error(INLINE_CELL_EDITING_ERROR);
+              }
+
+              const startEditCellCallback = () => startEditCells({
+                editingCells: [{ rowId, columnName }],
+              });
+              const newParams = startEditAction === 'click'
+                ? { ...params, onClick: startEditCellCallback }
+                : { ...params, onDoubleClick: startEditCellCallback };
+
+              return (
+                <TemplatePlaceholder
+                  params={{ ...newParams }}
```

Comment:
We have created a new object already. `params={{ ...newParams }}` -> `params={newParams}` ?